doc/source/guidelines/logging.rst

```diff
@@ -0,0 +1,306 @@
+Logging Guidelines
+##################
+
+This section describes several guidelines for logging in PyAnsys
+libraries. These guidelines are best practices discovered through
+implementing logging services and modules within PyAnsys
+libraries. Suggestions and improvements are welcome.
+
+
+Why and when to log
+===================
+Logging helps to track events occurring in the application.
+Whenever an information must be exposed, displayed and shared, logging is the
+way to do it.
+It is destinated to both the users and the application developers.
+It can serve several purposes:
+
+  - extract some valuable data for the final users to know the status of their work.
+  - track the progress and the course of the application usage.
+  - provide the developer with as much information as possible if an issue happens.
+
+The message logged can contain generic information or embed data specific
+to the current session.
+Message content is associated to a level of severity (info, warning, error...).
+Generally, this degree of significance indicates the recipient of the message.
+An info message is directed to the user while a debug message is useful for
+the developer itself.
+
+
+Logging in PyAnsys Libraries
+============================
+
+The logging capabilities in PyAnsys libraries should be built upon the
+`standard logging <https://docs.python.org/3/library/logging.html/>`__
+library.  PyAnsys libries should not to replace this library, rather provide
+a standardized way to interact between the built-in :mod:`logging`
+library and ``PyAnsys`` libraries.
+
```

Comment:
Add a best practices section and move the following sections here:
- Avoid printing to the console. Let's also place emphasis on using the built in `logging` module 
- Enable/Disable handlers (perhaps describe collection using `__del__` here)
- Formatting/filters

```suggestion

Logging Best Practices
----------------------

```

Additionally, I'd like to see the following sections:
- Links (at the top) to other blogs and other best practices. 
- Unit testing of logging (just provide a link and include the code in a "dropdown" section). For example:
```
.. collapse:: Example PyAnsys custom logger module

    .. literalinclude:: ../../../logging/pyansys_logging.py
```